homeassistant/components/flux_led/light.py

```diff
@@ -266,13 +266,21 @@ def effect(self):
 
     def _turn_on(self, **kwargs):
         """Turn the specified or all lights on."""
-        self._bulb.turnOn()
+        if not self.is_on:
+            self._bulb.turnOn()
+            if kwargs.get(ATTR_BRIGHTNESS) is None:
+                return
```

Comment:
This is wrong. I think you are addressing a UI issue but we can turn on with any parameters using `light.turn_on`.

Also, preferably turn on unconditionally because our local state could be stale.

I am not quite sure what you are trying to do so I cannot advise on a better way :)

Reply:
One (in my opinion) not so optimal thing is, that `turn_on` is for turning stuff on and changing the state of an entity. What I want to do is that when the lamp is off and the user want to turn it on, it just turns on and nothing else (except when brightness is set, because it's possible to turn the light on with the brightness slider as well). When the lamp is already on, then the state should be changed.

Reply:
Yes, `light.turn_on` is indeed overloaded but each light platform should not work around that. It is quite possible to turn on to an RGB color, for example with an automation.

Instead, only call `self._bulb.setRgb()` and friends if relevant `kwargs` are actually passed.

Reply:
Oh yeah, I forgot the automation, thank you. I will correct this.